lib/ansible/modules/system/authorized_key.py

```diff
@@ -534,6 +535,8 @@ def enforce_state(module, params):
                 module.fail_json(msg=error_msg % key)
             else:
                 key = resp.read()
+                if PY3:
+                    key = key.decode()
```

Comment:
We have three helper functions in ```ansible.module_utils._text```:
* ```to_text()```: transforms a string into the python version's text type (unicode on py2, str on py3)
* ```to_bytes()```: transforms the string into the python version's bytes type (str on py2 and bytes on py3)
* ```to_native()```: transforms the string into the python version's str type.

All three of these default to using utf-8 to encode strings and use surrogate_or_replace as their error handler (which means that if a byte sequence can't be encoded into utf-8/decoded from utf-8, the surrogateescape error handler will be used on python3 and a replacement character will be used on python2).

In module code, we make heavy use of to_native() as the modules have to work on both python2 and python3 and we want to continue to work with the many custom (not controlled by ansible) modules out there that use unmarked literals all over the place (like "Data café"... this will be a str on both python2 and python3.)  In controller-side code we're mostly able to use to_text() and to_bytes() to cleanly separate our bytes and text strings.